Backend/Microservices/AuthService/src/Auth.Application/Interfaces/IUserRepository.cs

```diff
@@ -0,0 +1,19 @@
+﻿using Auth.DAL.Entities;
+using Microsoft.AspNetCore.Identity;
+
+namespace Auth.BLL.Interfaces
+{
+    public interface IUserRepository
+    {
+        Task Create(User user, CancellationToken cancellationToken);
+        Task Delete(User user, CancellationToken cancellationToken);
+        Task<List<User>> Get(CancellationToken cancellationToken);
+        Task<User?> Get(long userId, CancellationToken cancellationToken);
+        Task<List<IdentityRole<long>>> GetRoles(User entity, CancellationToken cancellationToken);
```

Comment:
Pick one name for parameters (user or entity) and stick with it. Switching between them makes the code harder to follow.